pkg/apis/provisioning/v1alpha5/kubelet_configuration.go

```diff
@@ -22,4 +22,19 @@ type KubeletConfiguration struct {
 	// Note that not all providers may use all addresses.
 	//+optional
 	ClusterDNS []string `json:"clusterDNS,omitempty"`
+	// containerRegistry setting is optional and allows to configure image registry credentials.
+	//+optional
+	ContainerRegistry *ContainerRegistry `json:"containerRegistry,omitempty"`
```

Comment:
Thanks for the PR!

We want to limit the API surface that we expose on the Provisioner and adding every configuration possible for the kubelet or the container-runtime might cause difficulties with maintenance going forward. For example even here, you had to add support both with BottleRocket and the EKS Optimized AMIs but once we add Windows, it introduces another dimension in the matrix. 
We also need to be vendor neutral in the provisioning API, so everything that's added here ideally should apply across all cloud-providers.  

1. Once we get the PR for CustomUserData merged [`in for Bottlerocket`](https://github.com/aws/karpenter/pull/1720), I think we can rely on just setting the container-registry settings there. You could rebase this PR to include your changes in `bottlerocket.go` into `bottlerocketsettings.go`. 

2. For the AL2 AMI part, I think we want folks to update the kubeletConfig via UserData themselves. So a UserData like - 
```
MIME-Version: 1.0
Content-Type: multipart/mixed; boundary="//"

--//
Content-Type: text/x-shellscript; charset="us-ascii"
#!/bin/bash

KUBELET_CONFIG=/etc/kubernetes/kubelet/kubelet-config.json
echo "$(jq ".credentialProvider=...." $KUBELET_CONFIG)" > $KUBELET_CONFIG

--//--
```